c7n/resources/s3.py

```diff
@@ -2049,11 +2049,27 @@ def process_bucket(self, b):
         if found is False:
             self.log.debug("updating bucket:%s inventory configuration id:%s",
                            b['Name'], inventory_name)
-        client.put_bucket_inventory_configuration(
-            Bucket=b['Name'], Id=inventory_name, InventoryConfiguration=inventory)
+        try:
+            client.put_bucket_inventory_configuration(
```

Comment:
this would be a bit less code and better across errors kinds to do this as a worker catch around the concurrent futures stuff (ala submit instead  of map).

Reply:
ah true.  It actually looks like this should already be handled here: https://github.com/capitalone/cloud-custodian/blob/master/c7n/resources/s3.py#L1962 however ClientError exceptions are bubbling past this.

Reply:
oh wait that was for the filter.  I should be able to implement the same sort of logic in the action to prevent failures.